clients/venice-client/src/main/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClient.java

```diff
@@ -571,16 +570,22 @@ private void batchGetTransportRequestCompletionHandler(
     long totalDecompressionTimeForResponse = 0;
     VeniceCompressor compressor =
         metadata.getCompressor(transportClientResponse.getCompressionStrategy(), requestContext.currentVersion);
+    // Despite FastSerializerDeserializerFactory is caching deserializers by reader, writer and object class.
+    // Retrieving the corresponding deserializer still requires constructing some container object which could add up.
+    Map<Integer, RecordDeserializer<V>> recordDeserializerMap = new HashMap<>();
```

Comment:
can we make it class level variable?

Reply:
We cannot make it a class property if we lookup only by 1 ID (the reader schema), because the writer schema (which should be the latest schema) can change over time. If we want to make it a class property, then we need to look up by two IDs, and we already have a class tailor-made for this: `AvroStoreDeserializerCache`. This is likely the optimal approach, IMHO.

If we keep it within the scope of the function only, then we can lookup only by reader schema ID and assume the writer schema doesn't change much in the span of one function call. In that case it's ok to have it as a locally scoped variable. Though a `SparseConcurrentList` would be more optimal than a Integer-keyed map...

Reply:
Yeah exactly what @FelixGV said `getDataRecordDeserializer` gets the latest reader schema. If we cache it at the class level then even if the reader schema is updated it won't reflect until writer schema changes (triggers new compute if absent call) or app is bounced. That's different behavior than our thin-client today and could cause issues. 

@FelixGV you mean reader schema will not change in the span of one request call right? There could be different writer schemas in one call because keys are written at different times for a hybrid store.

Will checkout `SparseConcurrentList`.

Reply:
Yeah, just looked at `AvroStoreDeserializerCache` impl. It would be the better approach.

Reply:
`AvroStoreDeserializerCache` will work for generic store client and for specific store client, it may not work since the reader schema carried in SpecificRecord may not be same as the schemas registered in schema repo.

Reply:
@gaojieliu what is your recommendation?

Reply:
For Generic store client, it will just work and for specific store client, we can overload it by specifying the reader schema at construction time: `AvroSpecificStoreDeserializerCache`.
And that will potentially be easier to wrap up all the deserializer  cache related logics into separate class instead of repeating the same code in many different places.